src/iop/lut3d.c

```diff
@@ -583,8 +593,8 @@ setvbuf(stdout, NULL, _IONBF, 0);
   g->colorspace = dt_bauhaus_combobox_new(self);
   dt_bauhaus_widget_set_label(g->colorspace, NULL, _("application color space"));
   dt_bauhaus_combobox_add(g->colorspace, _("linear RGB"));
-  dt_bauhaus_combobox_add(g->colorspace, _("gamma sRGB"));
-  dt_bauhaus_combobox_add(g->colorspace, _("LAB"));
+  dt_bauhaus_combobox_add(g->colorspace, _("sRGB"));

```

Comment:
I don't think there should be any LUT produced on gamma-encoded data. If this is the case, usually the LUT is used to linearize the input.

Reply:
I don't understand here.
Gamma sRGB anf REC.709 are very close from each other. Same matrix, just the gamma curve is a bit different. A lot of LUT are given to be applied on REC.709 which is a gamma-encoded transformation by definition.

Reply:
REC709 defines the RGB primaries used in sRGB but has no gamma itself, it's a linear, scene-referred space. Gamma sRGB is just sRGB. Gamma REC709 can be sRGB. Now, HDTVs use REC709 primaries upon which they add a gamma, but that's none of our business in image processing.

You can't do linear alegebra on non-linear data, so applying an sRGB/sRGB LUT makes no sense, you need to revert the gamma, apply the LUT, and apply the gamma back. But I don't know any corporate LUT maker dumb enough to use sRGB. Usually, you want very large RGB spaces to apply LUTs.

Some LUTs can be applied on log-encoded video footage, and will therefore revert the log and remap the output to scene-referred. That's a particular way to bring back footage to scene-referred space.

But this does not affect us here, and more importantly, you don't want to correct colors on anything but linear. The LUT defines color samples for the remapping. For colors between the samples, an interpolation is used, if it is a linear interpolation and you are in log/gamma encoded in and out, the interpolation error will increase because it assumes the patches are evenly spaced, when gamma compresses highlights and raise midtones.

Really, we should remove any non-linear conversion from the pipeline and keep the gamma and stuff as the very last step before sending pixels to display. Gamma will mess-up every linear algebra you want to do: blur, rotation, color matching, etc.

Reply:
Our previous comments are related to display referenced LUT.
For those which are scene referenced, things are less clear for me. In particular for the log LUT. We have the option unbreak(log) + lut3d(prophotoregb). Is that the right option ? Any other way to rebuild the LOG colorspace ?

Reply:
> REC709 defines the RGB primaries used in sRGB but has no gamma itself, it's a linear, scene-referred space.

All the references I've found associate REC709 to TV display and give the definition with gamma.  Could you please provide some links where I could find better reference ?